libgadget/hydra.c

```diff
@@ -281,32 +296,34 @@ hydro_ngbiter(
         }
         double hfc_visc = 0.5 * P[other].Mass * visc * (dwk_i + dwk_j) / r;
         double hfc = hfc_visc;
-#ifndef DENSITY_INDEPENDENT_SPH
         double r1 = 1, r2 = 1;
-#else
-        double r1 = 0, r2 = 0;
-        /* leading-order term */
-        double EntOther = SPHP(other).EntVarPred;
-
-        hfc += P[other].Mass *
-            (dwk_i*iter->p_over_rho2_i*EntOther/I->EntVarPred +
-             dwk_j*p_over_rho2_j*I->EntVarPred/EntOther) / r;
-
-        /* enable grad-h corrections only if contrastlimit is non negative */
-        if(All.DensityContrastLimit >= 0) {
-            r1 = I->EgyRho / I->Density;
-            r2 = SPHP(other).EgyWtDensity / SPHP(other).Density;
-            if(All.DensityContrastLimit > 0) {
-                /* apply the limit if it is enabled > 0*/
-                r1 = DMIN(r1, All.DensityContrastLimit);
-                r2 = DMIN(r2, All.DensityContrastLimit);
+
+        if(All.DensityIndependentSphOn) {
+            /*This enables the grad-h corrections*/
+            r1 = 0, r2 = 0;
+            /* leading-order term */
+            double EntOther = SPHP(other).EntVarPred;
+
+            hfc += P[other].Mass *
+                (dwk_i*iter->p_over_rho2_i*EntOther/I->EntVarPred +
+                dwk_j*p_over_rho2_j*I->EntVarPred/EntOther) / r;
+
+            /* enable grad-h corrections only if contrastlimit is non negative */
+            if(All.DensityContrastLimit >= 0) {
+                r1 = I->EgyRho / I->Density;
+                r2 = SPHP(other).EgyWtDensity / SPHP(other).Density;
+                if(All.DensityContrastLimit > 0) {
+                    /* apply the limit if it is enabled > 0*/
+                    r1 = DMIN(r1, All.DensityContrastLimit);
+                    r2 = DMIN(r2, All.DensityContrastLimit);
+                }
             }
         }
-#endif
-        /* grad-h corrections: enabled if DENSITY_INDEPENDENT_SPH is off, or DensityConstrastLimit >= 0 */
+
+        /* grad-h corrections: enabled if DensityIndependentSphOn = 0 is off, or DensityConstrastLimit >= 0 */
```

Comment:
Typo = 0.

Reply:
DensityContractLimit caught my eye. It controls three behaviours if and only if DensityIndepdentSPH is enabled:
- > 0 Limit r1, and r2, enable the grad-h term
- = 0 no limit r1 and r2, enable the grad-h term
- < 0 disable the grad-h term.

Could you also update the doc string of this in params.c?

Reply:
Done!